### PR TITLE
Tweak the scons help message

### PR DESCRIPTION
**Pull request by mwichmann**
* <s>Processing changed a bit - some lines can now be joined instead of split (if the usage part is short so the combination will still fit).</s> - disabled
* Dropped the "Ingored for compatibility" chunk from the printout.
* Some of the help texts were changed slightly - a consistent display for available-choices, and ending periods dropped (these aren't complete sentences anyway)
* The usage: message was changed as it didn't mention variables.  A number of tests expected the exact value of that line, and so were updated.
* Updated docstrings (for the API docs).
* Formatter is primed with detected terminal width, so it uses a wider (or narrower) terminal, not just 80-column

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [X] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -46,6 +46,7 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
       Calls now look like 'debug("template %s", text)' rather than
       'debug("template %s" % text)' so the logging system does the
       interpolation only when/if needed (was a pylint warning).
+    - Update Help display a bit - join some lines, drop "ignored for compat".
```
Comment (bdbaddog) [inline CHANGES.txt]: Perhaps add
(help output with command line option `-H`)

**Comment by bdbaddog** [comment]
I'm not sure I like these changes.

```
  --cache-debug=FILE          Print CacheDir debug info to FILE.
  --cache-disable, --no-cache  Do not retrieve built targets from CacheDir.
  --cache-force, --cache-populate  Copy already-built targets into the CacheDir.
```

vs

```
  --cache-debug=FILE          Print CacheDir debug info to FILE.
  --cache-disable, --no-cache
                              Do not retrieve built targets from CacheDir.
  --cache-force, --cache-populate
                              Copy already-built targets into the CacheDir.
```

I don't think that's really an improvement, now the formatting is not even..

What might be of value is to detect the width of the terminal and set that as the width, rather than artificially setting width at 80 chars?

Also not sure it's a good idea to hide the flags which are ignored to maintain make command line flag compat.

**Comment by mwichmann** [comment]
Getting into terminal-size fiddling isn't easy and I think not  reliably cross-platform. Update: looks like I'm wrong, there's `shutil.get_terminal_size()`, which even seems to work on Windows.

The folding or not is a matter of taste, obviously, I think the folding to a new line when not needed (esp. the cache-disable example above) is silly, but I can't really argue.

**Comment by mwichmann** [comment]
The terminal-size detection is added in a new commit.

File: CHANGES.txt (hunk 1 above)
Comment (mwichmann) [reply CHANGES.txt]: I updated this line (there was also a new change, which was added to this entry)

**Comment by bdbaddog** [comment]
Please add a blurb to RELEASE.txt and we're good to go!

**Comment by mwichmann** [comment]
For which - the terminal width?

**Comment by bdbaddog** [comment]
> For which - the terminal width?

basically just what you had in CHANGES.txt

```
Update Help (-H) output a bit. Drop "ignored for compat" entry.
      Pass window size to formatter so it formats for wider displays too.
```

**Comment by mwichmann** [comment]
> > For which - the terminal width?
> 
> basically just what you had in CHANGES.txt

done (reworded a bit)

**Pull request merged**